src/gsy_e/models/strategy/load_hours.py

```diff
@@ -51,109 +51,140 @@
+
+    def _assign_hours_of_day(self, hrs_of_day: List[int]):
         if hrs_of_day is None:
             hrs_of_day = list(range(24))
 
-        # be a parameter on the constructor or if we want to deal in percentages
-        if hrs_per_day is None:
-            hrs_per_day = len(hrs_of_day)
-
         self.hrs_of_day = hrs_of_day
-        self._initial_hrs_per_day = hrs_per_day
 
         if not all(0 <= h <= 23 for h in hrs_of_day):
             raise ValueError("Hrs_of_day list should contain integers between 0 and 23.")
 
-        if len(hrs_of_day) < hrs_per_day:
-            raise ValueError("Length of list 'hrs_of_day' must be greater equal 'hrs_per_day'")
 
-    def add_entry_in_hrs_per_day(self, current_day: int, overwrite: bool = False) -> None:
-        """Add the current day (in simulation) with the mapped hrs_per_day."""
-        if current_day not in self.hrs_per_day or overwrite:
-            self.hrs_per_day[current_day] = self._initial_hrs_per_day
+class LoadHoursPerDayEnergyParameters(LoadHoursEnergyParameters):
+    """Add the hours-per-day quota parameter to the LoadHoursEnergyParameters."""
+    def __init__(self, avg_power_W, hrs_per_day=None, hrs_of_day=None):
+        LoadValidator.validate_energy(
+            avg_power_W=avg_power_W, hrs_per_day=hrs_per_day, hrs_of_day=hrs_of_day)
 
-    def reset(self, current_day: int, **kwargs) -> None:
-        """Reset strategy parameters."""
-        if (kwargs.get("hrs_per_day") is not None or
-                kwargs.get("hrs_of_day") is not None):
-            self._assign_hours_of_per_day(kwargs["hrs_of_day"], kwargs["hrs_per_day"])
-            self.add_entry_in_hrs_per_day(
-                current_day, overwrite=True
-            )
-        if kwargs.get("avg_power_W") is not None:
-            self.avg_power_W = kwargs["avg_power_W"]
+        super().__init__(avg_power_W, hrs_of_day)
 
-    def set_energy_measurement_kWh(self, time_slot: DateTime) -> None:
-        """Set the (simulated) actual energy consumed by the device in a market slot."""
-        energy_forecast_kWh = self.state.get_desired_energy_Wh(time_slot) / 1000
-        simulated_measured_energy_kWh = utils.compute_altered_energy(energy_forecast_kWh)
+        # Maps each simulation day to the number of active hours in that day
+        self.hrs_per_day: Dict[int, int] = {}
+        self._initial_hrs_per_day: Optional[int] = None
+        self._assign_hours_per_day(hrs_per_day)
 
-        self.state.set_energy_measurement_kWh(simulated_measured_energy_kWh, time_slot)
+    def serialize(self):
+        return {
+            **super().serialize(),
+            "hrs_per_day": self.hrs_per_day,
+        }
 
-    def allowed_operating_hours(self, time):
-        """Check if timeslot inside allowed operating hours."""
-        return time.hour in self.hrs_of_day
+    def _assign_hours_per_day(self, hrs_per_day: int):
+        # be a parameter on the constructor or if we want to deal in percentages
```

Comment:
Sorry, I do not get this comment. What do you mean by `if we want to deal in percentages` ?

Reply:
This is probably a trail from moving code. Removing this comment, thanks for catching it!